### PR TITLE
Update Eigen version to commit e5e305a158a029f5b5f837bf821411a51439a970.

### PR DESCRIPTION
**Pull request by rmlarsen**
PiperOrigin-RevId: 201624024

### CODE REVIEW

**Review by gunan — APPROVED** [review]

File: tensorflow/workspace.bzl [hunk 1]
```diff
@@ -107,11 +107,11 @@ def tf_workspace(path_prefix="", tf_repo_name=""):
...
-          "https://bitbucket.org/eigen/eigen/get/6913f0cf7d06.tar.gz",
+          "https://mirror.bazel.build/bitbucket.org/eigen/eigen/get/e5e305a158a0.tar.gz",
+          "https://bitbucket.org/eigen/eigen/get/e5e305a158a0.tar.gz",
       ],
-      sha256 = "791b836cacd03e20bae5bdd25f1c4a5505a0a9975ba94a61eb4e2631fbd1d53a",
-      strip_prefix = "eigen-eigen-6913f0cf7d06",
+      sha256 = "8bbe676d69e7f59070c83a949454b8b6344034e0ebbf686b337528e5dc04c7de",
+      strip_prefix = "eigen-eigen-e5e305a158a0",
       build_file = clean_dep("//third_party:eigen.BUILD"),
       patch_file = clean_dep("//third_party:eigen_fix_cuda_compilation.patch")
```
Comment (gunan) [inline tensorflow/workspace.bzl]: I don't think we still need this patch. Could you try removing this line? (If you can, removing corresponding file will also be very helpful)

**Comment by wdirons** [comment]
This level of eigen contains a typo that fails the compile on ppc64le. Is it possible to not merge this into 1.9 until the typo is fixed? I can create a PR for the master branch when it is fixed.

The bug is this commit: https://bitbucket.org/eigen/eigen/commits/d2f8e1b572c0/
They are missing the closing ')'. I requested an account at eigen so I can comment on the bug. (Trying to get this fixed as soon as possible)

**Comment by rmlarsen** [comment]
@wdirons we really should have this change in release R1.9. It will be a major performance improvement. However, I can submit a PR to Eigen to fix the issue you mention first.

**Comment by wdirons** [comment]
Thank you @rmlarsen , I also got my team mate to add a comment to the bug: http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1555

I'll keep an eye on it for a fix.

**Comment by rmlarsen** [comment]
Upstream Eigen PR: https://bitbucket.org/eigen/eigen/pull-requests/410/fix-typo-in-pbend-for-altivec/diff

**Comment by wdirons** [comment]
@rmlarsen , your patch was merged and I verified it on ppc64le. Thank you!

PR #20254 created to update Eigen in the master branch.

**Comment by case540** [comment]
Re-triggering Windows Bazel build for a third time. Not sure why it keeps on failing, but I feel its unlikely due to this change.

**Comment by gunan** [comment]
@case540 ready to merge when the tests pass.

**Pull request merged**